bids2openminds/main.py

```diff
@@ -2,18 +2,70 @@
 import os
 import pathlib
 from warnings import warn
-
+from bids import BIDSLayout
 import pandas as pd
 from nameparser import HumanName
 
 import openminds.latest.core as omcore
 import openminds.latest.controlled_terms as controlled_terms
 from openminds import IRI
 
-from .utility import table_filter, pd_table_value, file_hash, file_storage_size
+from .utility import table_filter, pd_table_value, file_hash, file_storage_size, read_json
 from .mapping import bids2openminds_instance
 
 
+def read_bids_directory(input_path):
+    """
+    Reads the content of the dataset_description.json file and determines whether the dataset is raw, derivative, or contains derivatives. Then, it calls the appropriate BIDSLayout function.
+    Returns three Python objects.
+
+    Parameters:
+    - input_path: The path to the BIDS folder.
+
+    Returns:
+    - bids_layout: An instance of the Layout class from pybids for this dataset.
+    - layout_df: A pandas DataFrame containing all the files and corresponding information.
+    - dataset_description: A Python dictionary containing the content of the dataset_description JSON file.
+    """
+
+    if not os.path.isdir(input_path):
+        raise NotADirectoryError(
+            f"The input path is not valid. You specified '{input_path}', which is not a directory.")
+
+    # if not(BIDSValidator().is_bids(input_path)):
+    #  raise NotADirectoryError(f"The input directory is not valid, you have specified {input_path} which is not a BIDS directory.")
+    dataset_description_path = os.path.join(
+        input_path, "dataset_description.json")
+    dataset_description = read_json(dataset_description_path)
+
+    if not dataset_description:
+        bids_layout = BIDSLayout(input_path)
+        layout_df = bids_layout.to_df()
+        dataset_description_path = table_filter(layout_df, "description")
+        dataset_description = read_json(dataset_description_path.iat[0, 0])
+        if not dataset_description:
+            raise FileNotFoundError(
+                "There was no dataset_description.json file. Every dataset MUST include this file."
+            )
+        return bids_layout, layout_df, dataset_description
+
+    if ("DatasetType" in dataset_description) and (dataset_description["DatasetType"] == "derivative"):
```

Comment:
since `dataset_description["DatasetType"]` is only RECOMMENDED this is not a reliable path to check for derivatives. 

I would suggest to only go with line 57-66 to determine if derivatives are present. This only works though in the case of derivatives that are provided in a subdirectory of a BIDS raw dataset. In case only derivative data are shared as BIDS dataset, there is no requirement to store those data in a "derivative" subdirectory (if I understand the specifications correctly).

However you can make use of the REQUIREMENT that the dataset_description.json MUST have a `"GeneratedBy"` key specified. 

Also note that if derived datasets are stored in a subdirectory of a raw dataset in BIDS, there are two dataset_description.json files: one for the raw data and one for (each) derived dataset.